fairseq/data/data_utils.py

```diff
@@ -227,10 +243,25 @@ def batch_by_size(
 
     if isinstance(indices, types.GeneratorType):
         indices = np.fromiter(indices, dtype=np.int64, count=-1)
-
     return batch_by_size_fast(indices, num_tokens_fn, max_tokens, max_sentences, bsz_mult)
 
 
+def batch_by_size_tpu(
+    indices, num_tokens_fn, input_shapes
+):
+    batches = [[] for _ in input_shapes]
```

Comment:
I don't think that works because

```
>>> x=[[]]*10
>>> x
[[], [], [], [], [], [], [], [], [], []]
>>> x[0].append(5)
>>> x
[[5], [5], [5], [5], [5], [5], [5], [5], [5], [5]]
```

Reply:
Oh right!